lib/streamlit/hashing.py

```diff
@@ -242,6 +242,15 @@ def _update(self, hasher, obj, context=None):
         b = self.to_bytes(obj, context)
         hasher.update(b)
 
+    def _file_should_be_hashed(self, filename):
+        filepath = os.path.abspath(filename)
+        file_is_local = file_util.file_is_in_folder_glob(
+            filepath, self._get_main_script_directory()
+        )
+        file_is_in_pythonpath = file_util.file_in_pythonpath(filepath)
+        file_is_blacklisted = self._folder_black_list.is_blacklisted(filepath)
+        return (file_is_local or file_is_in_pythonpath) and not file_is_blacklisted
```

Comment:
Nit - this _probably_ doesn't make a difference, but it feels a bit wasteful to be calculating all these values all the time. @tvst may totally disagree with this, but I'd do something like:

```python
if self._folder_black_list.is_blacklisted(filepath):
    return False

return (
    # Local file?
    file_util.file_is_in_folder_glob(filepath, self._get_main_script_directory()) or
    # File in python path?
    file_util.file_in_pythonpath(filepath)
)
```

Reply:
yeah... you are right, still, impacts readability...

Reply:
Ok, I am doing the short circuiting here...